src/functions/severa/get-filtered-workhours/handler.ts

```diff
@@ -11,56 +11,64 @@ import type SeveraResponseWorkHours from "src/types/severa/workHour/severaRespon
  * @param event - API Gateway event containing the userId, projectId, and phaseId.
  */
 export const getWorkHoursHandler: APIGatewayProxyHandler = async (event) => {
-  const { startDate, endDate, severaProjectId, severaPhaseId, severaUserId } = event.queryStringParameters || {};
+  const { startDate: startDate, endDate: endDate, severaProjectId, severaPhaseId, severaUserId } = event.queryStringParameters || {};
 
   try {
     const api = CreateSeveraApiService();
+    // Getting the list of opted in users
+    const optInUsers = await api.getOptInUsers();
+    const optInUserGuids = new Set(optInUsers.map((u) => u.guid));
 
-    /**
-     * Construct a custom url for fetching workHours with required queryParams.
-     *
-     * @param severaProjectId - Severa project id
-     * @param severaUserId - Severa user id
-     * @param startDate - Start date for work hours
-     * @param endDate - End date for work hours
-     *
-     * @returns {string} - The constructed custom URL for fetching work hours.
-     */
-    const buildWorkHoursUrl = (severaProjectId?: string, severaUserId?: string, startDate?: string, endDate?: string) => {
-      let endpointPath: string;
-
-      if (severaProjectId) {
-        endpointPath = `projects/${severaProjectId}/workhours`;
-      } else if (severaUserId) {
-        endpointPath = `users/${severaUserId}/workhours`;
+    let errors: Array<{ userGuid: string, error: any }> = [];
+    // If severaProjectId is specified and severaUserId is not, make requests for each opted-in user
+    let allWorkHours: SeveraResponseWorkHours[] = [];
+    if (severaProjectId) {
+      // Request in parallel for each opted-in user
+      const workHoursResults = await Promise.all(
+        optInUsers.map(async (user) => {
+          try {
+            const url = new URL(`${process.env.SEVERA_DEMO_BASE_URL}/v1/users/${user.guid}/workhours`);
+            // This endpoint/query parameter is working but it is not documented proper in Severa API spec
+            url.searchParams.append("projectGuid", severaProjectId);
+            if (startDate) url.searchParams.append("startDate", startDate);
+            if (endDate) url.searchParams.append("endDate", endDate);
+            const userWorkHours = await api.getWorkHours(url);
+            return userWorkHours;
+          } catch (e) {
+            errors.push({ userGuid: user.guid, error: e });
+            return [];
+          }
+        })
+      );
+      // Combine all results
+      allWorkHours = workHoursResults.flat();
+      // Filtering only for the specified project
```

Comment:
```suggestion
      // Filtering only for the specified project
```
Remove unnecessary comment